### PR TITLE
all: remove wgcfg.IP and wgcfg.CIDR

### PR DESCRIPTION
**Pull request by josharian**
Replace them with inet.af/netaddr.

This is a first step towards eliminating wgcfg,
which is a prerequisite of eliminating this fork.

### CODE REVIEW

File: device/config.go [hunk 1]
```diff
@@ -39,11 +40,10 @@ func (device *Device) Config() *wgcfg.Config {
 			p.Endpoints = peer.endpoint.Addrs()
 		}
 		for _, ipnet := range device.allowedips.EntriesForPeer(peer) {
-			ones, _ := ipnet.Mask.Size()
-			cidr := wgcfg.CIDR{
-				Mask: uint8(ones),
+			cidr, ok := netaddr.FromStdIPNet(&ipnet)
+			if !ok {
+				panic("bad ipnet " + ipnet.String())
```
Comment (danderson) [inline device/config.go:45]: Eek, panic? No way to be more graceful?

File: device/config.go [hunk 2]
```diff
@@ -166,8 +166,8 @@ func (device *Device) Reconfig(cfg *wgcfg.Config) (err error) {
 		// Mask) are overwritten on every iteration through the
 		// loop. The loop owns its memory; don't retain references into it.
 		for _, allowedIP := range p.AllowedIPs {
-			ones := uint(allowedIP.Mask)
-			ip := allowedIP.IP.IP()
+			ones := uint(allowedIP.Bits)
+			ip := allowedIP.IP.IPAddr().IP
```
Comment (danderson) [inline device/config.go:170]: Huh, we're lacking a StdIP() helper to avoid the indirection through IPAddr. Maybe that was deliberate, dunno.

**Review by danderson — APPROVED** [review]

File: device/config.go (hunk 1 above)
Comment (josharian) [reply device/config.go]: Done.

File: device/config.go (hunk 2 above)
Comment (josharian) [reply device/config.go:171]: Yeah, I'm not sure. It didn't show up often enough for me to be bothered by.

Comment (bradfitz) [reply device/config.go:171]: It was intentional. IPAddr is a superset of IP anyway and it's just as easy to write `.IP` to get just the IP when needed. That limits how much compat API is needed in netaddr.

**Comment by bradfitz** [comment]
yay!

**Pull request merged**